smart_contracts/savings-pool.py

```diff
@@ -0,0 +1,4564 @@
+import smartpy as sp
+
+Constants = sp.import_script_from_url("file:common/constants.py")
+Token = sp.import_script_from_url("file:token.py")
+
+################################################################
+################################################################
+# STATE MACHINE STATES
+################################################################
+################################################################
+
+IDLE = 0
+WAITING_UPDATE_BALANCE = 1
+WAITING_REDEEM = 2
+WAITING_DEPOSIT = 3
+
+################################################################
+################################################################
+# CONTRACT
+################################################################
+################################################################
+
+Addresses = sp.import_script_from_url("file:./test-helpers/addresses.py")
+Errors = sp.import_script_from_url("file:./common/errors.py")
+FA12 = sp.import_script_from_url("file:./fa12.py")
+
+class SavingsPoolContract(FA12.FA12):
+  def __init__(
+    self,
+    
+    # The address of the token contract which will be deposited.
+    tokenContractAddress = Addresses.TOKEN_ADDRESS,
+
+    # The governor of the pool.
+    governorContractAddress = Addresses.GOVERNOR_ADDRESS,
+
+    # The address of the stability fund.
+    stabilityFundContractAddress = Addresses.STABILITY_FUND_ADDRESS,
+
+    # The interest rate.
+    interestRate = sp.nat(0),
+
+    # The last time the interest rate was updated.
+    lastInterestCompoundTime = sp.timestamp(0),
+
+    # The initial token balance.
+    underlyingBalance = sp.nat(0),
+
+    # The initial state of the state machine.
+    state = IDLE,
+
+    # State machine states - exposed for testing.
+    savedState_tokensToRedeem = sp.none,
+    savedState_redeemer = sp.none,
+    savedState_tokensToDeposit = sp.none,
+    savedState_depositor = sp.none,
+  ):
+    token_id = sp.nat(0)
+
+    token_entry = sp.map(
+      l = {
+        "name": sp.bytes('0x496e7465726573742042656172696e67206b555344'), # Interest Bearing kUSD
+        "decimals": sp.bytes('0x3138'), # 18
```

Comment:
Also is this actually an 18 digit mantissa? I just got back something with 36 decimals 🤔